src/lib/methods/batch/batch-item.ts

```diff
@@ -0,0 +1,7 @@
+export abstract class BatchItem {
+
+	protected constructor(public table: string, public key: any) {
+	}
+
+	abstract get value();
```

Comment:
I would make this `abstract buildRawQuery(): BatchItemQuery` where `BatchItemQuery` is of type

```ts
interface BatchItemQuery {
	PutRequest: {
		Item: {
			[key: string]: any;
		};
	};
	DeleteRequest: {
		Item: {
			[key: string]: any;
		};
	};
}
```

This is more in line with how all the other query stuff works. All methods just have a `buildRawQuery()` which returns the query from that item.